### PR TITLE
Fix RSS import

### LINKED ISSUE #8867: New Medium stories are not fetched as drafts
**To Reproduce**

1. Click on your profile pic to open the drop down menu
2. Choose settings
3. Choose Publishing from RSS
4. Make sure your Medium feed is defined as follows: _https://medium.com/feed/@your_username_
My rss feed: https://medium.com/feed/@dvirsegal
5. Click on "Fetch feed now" and wait to "Your profile was successfully updated." message
6. Go to dashboard by either click on your profile drop down menu or from rss page hyperlink
 
**Expected behavior**

The new article is fetched and shown as draft

**Desktop (please complete the following information):**

- OS: windows 10
- Browser: Brave \ Chrome
- Version: Version 1.10.93 Chromium: 83.0.4103.106 (Official Build) (64-bit)

**Smartphone (please complete the following information):**

- Device: OP3
- OS: Android 10
- Browser: Brave
- Version: 1.10.94

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Bug Fix

## Description

We received a report about RSS import from a Medium feed not importing new articles. I tried it out locally and noticed a few things:

* We try to import them, but several posts raise an exception and get ignored. Maybe we should just silently swallow these (`app/services/rss_reader.rb:43`), but show a flash message like "Imported with errors" or somethjing instead. That's for another time though.
* `RssReeder#log_error` uses uses `Rails.logger.error` with two arguments (message and metadata), which itself raises an exception:
    ```ruby
    Rails.logger.error("test", {})
    #=> ArgumentError: wrong number of arguments (given 2, expected 0..1)
    ```
    This is due to Timber, which is only enabled in the production environment (https://github.com/thepracticaldev/dev.to/blob/7536ad293f832767a14213c827a534fcfb33deb8/config/environments/production.rb#L102-L106)
* `RssReader::Assembler#parse_and_translate_tweet!` raised an error since the blockquote markup doesn't look like our Nokogiri parsing code expect it to. Again, I'm not sure if anything changed there (e.g. in how Medium embeds tweets), but I updated this code to something that potentially/hopefully works a bit more reliably.

## Related Tickets & Documents

Closes #8867 

## QA Instructions, Screenshots, Recordings

_Please replace this line with instructions on how to test your changes, as well
as any relevant images for UI changes._

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [ ] docs.dev.to
- [ ] readme
- [ ] no documentation needed

## [optional] What gif best describes this PR or how it makes you feel?

alt_text

### CODE REVIEW

File: app/services/rss_reader.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def create_articles_for_user(user)
       articles.append(article)
     rescue StandardError => e
       log_error(
-        "RssReaderError: occurred while creating article",
+        "RssReaderError: occurred while creating article #{item.url}",
```
Comment (citizen428) [inline app/services/rss_reader.rb:43]: This is left over from my debugging, I can remove it later if you want.

File: app/services/rss_reader.rb [hunk 2]
```diff
@@ -121,7 +121,8 @@ def article_exists?(user, item)
     relation.where(title: title).or(relation.where(feed_source_url: feed_source_url)).exists?
   end
 
-  def log_error(error_msg, metadata)
-    Rails.logger.error(error_msg, metadata)
+  # TODO: [@thepracticaldev/oss]: How did the metadata argument to Rails.logger work before?
```
Comment (citizen428) [inline app/services/rss_reader.rb:124]: See the PR description. I didn't want to spend too much time on this, so just removed metadata logging for now (which is also the reason for the extra debug output above).

File: app/services/rss_reader/assembler.rb [hunk 3]
```diff
@@ -96,10 +96,10 @@ def parse_and_translate_tweet!(html_doc)
         bq_with_p = bq.css("p")
         next if bq_with_p.empty?
 
-        second_content = bq_with_p.css("p")[1].css("a")[0].attributes["href"].value
```
Comment (citizen428) [inline app/services/rss_reader/assembler.rb:99]: I'm not sure if the structure of this markup changed on Medium, but there's no more `p` tag inside `bq_with_p` only the `a` tag.

File: app/services/rss_reader/assembler.rb [hunk 4]
```diff
@@ -96,10 +96,10 @@ def parse_and_translate_tweet!(html_doc)
...
-        if bq_with_p.length == 2 && second_content.include?("twitter.com")
+        if (tweet_link = bq_with_p.css("a[href*='twitter.com']"))
```
Comment (citizen428) [inline app/services/rss_reader/assembler.rb:99]: Changed the selector to match Twitter URLs instead of doing that separately with `.include?`.

**Comment by citizen428** [comment]
This PR doesn't fix everything so far, I now get validation errors for 2 items regarding missing titles, which is next on my list. 

Also @rhymes in the meantime confirmed that the `Rails.logger` issue is "thanks" to Timber.

**Comment by citizen428** [comment]
I have to go offline now. If someone wants to take over, the remaining problem is validation errors (`Bad Authentication data.`) on one or two feed items. I'm using the feed from the issue for testing (https://medium.com/feed/@dvirsegal) and the feed item for this URL is problematic: https://medium.com/@dvirsegal/diving-into-the-deep-code-waters-574b95c8c3cb is one of the problematic posts

File: app/services/rss_reader.rb (hunk 1 above)
Comment (mstruve) [reply app/services/rss_reader.rb:43]: All this logging will be getting an overhaul soon, you can leave it

**Comment by DeJaVoo** [comment]
Just out of curiosity, since I opened New Medium stories are not fetched as drafts #8867 , when this fix will go live?

**Comment by maestromac** [comment]
@DeJaVoo This is now live. You can force pull in https://dev.to/settings/publishing-from-rss or give it ~20 min to see if it's coming through

**Pull request merged**